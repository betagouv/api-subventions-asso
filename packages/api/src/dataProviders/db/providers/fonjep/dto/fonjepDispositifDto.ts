export default interface FonjepDispositifDto {
    ID: number;
    Libelle: string;
    FinanceurCode: string;
}
