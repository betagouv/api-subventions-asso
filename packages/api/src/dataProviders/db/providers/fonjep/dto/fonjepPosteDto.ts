export default interface FonjepPosteDto {
    [key: string]: any;
    // dates are excel dates (i.e. days since 1900-01-01)
    Code: string | null;
    DispositifId: number | null;
    PstStatutPosteLibelle: string | null;
    PstRaisonStatutLibelle: string | null;
    FinanceurPrincipalCode: string | null;
    FinanceurAttributeurCode: string | null;
    AssociationBeneficiaireCode: string | null;
    AssociationImplantationCode: string | null;
    Annee: number | null;
    MontantSubvention: number | null;
    DateFinTriennalite: number | null;
    PstTypePosteCode: string | null;
    PleinTemps: string | null;
    DoublementUniteCompte: string | null;
}
