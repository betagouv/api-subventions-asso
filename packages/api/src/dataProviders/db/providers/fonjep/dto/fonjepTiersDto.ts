export default interface FonjepTiersDto {
    [key: string]: any;

    Code: string | null;
    RaisonSociale: string | null;
    EstAssociation: string | null;
    EstCoFinanceurPostes: string | null;
    EstFinanceurPostes: string | null;
    SiretOuRidet: string | null;
    CodePostal: string | null;
    Ville: string | null;
    ContactEmail: string | null;
}
