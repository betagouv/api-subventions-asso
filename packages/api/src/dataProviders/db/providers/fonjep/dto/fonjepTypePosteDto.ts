export default interface FonjepTypePosteDto {
    Code: string | null;
    Libelle: string | null;
}
