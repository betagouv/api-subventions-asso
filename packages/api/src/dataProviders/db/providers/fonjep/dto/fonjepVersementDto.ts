// null represents an empty cell in the file
export default interface FonjepVersementDtoWithExcelDate {
    PosteCode: string;
    PeriodeDebut: number | null; // dates in excel dates (i.e. days since 1900-01-01)
    PeriodeFin: number | null; // dates in excel dates (i.e. days since 1900-01-01)
    DateVersement: number | null; // dates in excel dates (i.e. days since 1900-01-01)
    MontantAPayer: number | null;
    MontantPaye: number | null;
}

export interface FonjepVersementDto
    extends Omit<FonjepVersementDtoWithExcelDate, "DateVersement" | "PeriodeDebut" | "PeriodeFin"> {
    PeriodeDebut: Date | null;
    PeriodeFin: Date | null;
    DateVersement: Date | null;
}
