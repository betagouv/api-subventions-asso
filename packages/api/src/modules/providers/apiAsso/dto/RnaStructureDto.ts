export interface RnaStructureDto {
    identite: {
        nom: string;
        sigle: string;
        id_rna: string;
        date_pub_jo: string;
        date_modif_rna: string;
        active: boolean;
        nature: string;
    };
    activites: {
        objet: string;
        lib_objet_social1: string;
        annee_activite_principale: number;
        effectif_salarie_cent: number;
        annee_effectif_salarie_cent: number;
    };
    coordonnees: {
        adresse_siege: {
            cplt_1: string;
            num_voie: number;
            voie: string;
            commune: string;
            cp: number;
            bp?: string;
            type_voie: string;
        };
        adresse_gestion: {
            cplt_1: string;
            voie: string;
            commune: string;
            cp: number;
            pays: string;
        };
    };
}
