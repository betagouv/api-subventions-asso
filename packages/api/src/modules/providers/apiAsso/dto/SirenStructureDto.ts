export interface SirenStructureEtablissementDto {
    actif: boolean;
    adresse: {
        cplt_1?: string;
        num_voie: number;
        voie: string;
        commune: string;
        code_insee: number;
        cp: number;
        type_voie: string;
    };
    id_siret: number;
    est_siege: boolean;
    date_actif: string;
    id_activite_principale: string;
    annee_activite_principale: number;
    effectif_salarie_cent: number;
    id_tranche_effectif?: string;
    annee_effectif_salarie_cent: number;
}

export interface SirenStructureDto {
    identite: {
        nom: string;
        nom_sirene: string;
        id_siren: number;
        id_siret_siege: number;
        id_forme_juridique: number;
        date_creation_sirene: string;
        date_modif_siren: string;
        active: boolean;
        active_sirene: boolean;
        regime: string;
    };
    activites: {
        id_activite_principale: string;
        annee_activite_principale: number;
        id_tranche_effectif: string;
        effectif_salarie_cent: number;
        annee_effectif_salarie_cent: number;
        appartenance_ess: string;
    };
    coordonnees: {
        adresse_siege: {
            cplt_1: string;
            num_voie: number;
            voie: string;
            commune: string;
            code_insee: number;
            cp: number;
            type_voie: string;
        };
        adresse_gestion: {
            cplt_1: string;
            num_voie: number;
            voie: string;
            commune: string;
            code_insee: number;
            cp: number;
            type_voie: string;
        };
    };
    nbEtabsActifs: number;
    etablissement?: SirenStructureEtablissementDto[];
}
