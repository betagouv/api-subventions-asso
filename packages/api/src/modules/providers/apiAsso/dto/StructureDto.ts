import { Rna } from "dto";

export interface StructureEtablissementDto {
    id_siret: number;
    id_siren: number;
    actif: boolean;
    date_actif: string;
    est_siege: boolean;
    nom?: string;
    telephone?: string;
    courriel?: string;
    note?: string;
    id_activite_principale: string;
    lib_activite_principale: string;
    annee_activite_principale: number;
    id_tranche_effectif: string;
    lib_tranche_effectif: string;
    effectif_salarie_cent: string;
    annee_effectif_salarie_cent: number;
    adresse: {
        cplt_1?: string;
        cplt_2?: string;
        cplt_3?: string;
        num_voie?: number;
        type_voie?: string;
        voie?: string;
        bp?: string;
        cedex?: string;
        cp?: number;
        commune?: string;
        code_insee?: number;
        pays?: string;
    };
}

export interface StructureRibDto {
    titulaire: string;
    banque: string;
    domiciliation: string;
    iban: string;
    bic: string;
    deleted: boolean;
    id_siret: number;
    url?: string;
}

export interface StructureRepresentantLegalDto {
    civilit√©: string;
    nom: string;
    prenom: string;
    fonction: string;
    est_representant_legal: boolean;
    valideur_cec: boolean;
    publication_internet: boolean;
    telephone: string;
    courriel: string;
    deleted: boolean;
    id_siret: number;
}

export interface StructureRnaDocumentDto {
    id: string;
    type: string;
    sous_type: string;
    lib_sous_type: string;
    annee?: number;
    time?: number;
    url: string;
}

export interface StructureDacDocumentDto {
    uuid: string;
    origin: string;
    nom: string;
    commentaire: string;
    annee_validite: number;
    time_depot: string;
    url: string;
    etat: string;
    meta: {
        type: string;
        id_siret: number;
        etat: string;
        iban?: string;
    };
}

export interface StructureDocumentDto {
    asso: {
        documents: DocumentsDto;
    };
}

export interface DocumentsDto {
    nbDocDac: number;
    nbDocRna: number;
    document_dac: StructureDacDocumentDto[];
    document_rna: StructureRnaDocumentDto[];
}

export default interface StructureDto {
    identite: {
        nom: string;
        nom_sirene?: string;
        sigle: string;
        sigle_sirene?: string;
        id_rna: Rna;
        id_ex: string;
        id_siren?: number;
        id_siret_siege: number;
        id_correspondance: string;
        id_forme_juridique: string;
        lib_forme_juridique: string;
        regime: string;
        date_pub_registre?: string;
        volume?: string;
        folio?: string;
        tribunal_instance?: string;
        date_pub_jo?: string;
        date_creat: string;
        date_creation_sirene: string;
        date_dissolution: string;
        date_modif_rna: string;
        date_modif_siren: string;
        nature: string;
        active: boolean;
        util_publique: boolean;
        date_publication_util_publique: string;
        groupement: string;
        domaine: string;
        type_structure_sportive?: string;
        eligibilite_cec: boolean;
        raison_non_eligibilite_cec: string;
        impots_commerciaux: boolean;
    };
    activites: {
        objet: string;
        id_objet_social1: string;
        lib_objet_social1: string;
        id_objet_social2: string;
        lib_objet_social2: string;
        id_activite_principale: string;
        lib_activite_principale: string;
        annee_activite_principale: number;
        champ_action_territorial: string;
        id_tranche_effectif: string;
        lib_tranche_effectif: string;
        effectif_salarie_cent: string;
        annee_effectif_salarie_cent: number;
        appartenance_ess: string;
        date_appartenance_ess: string;
        volontaire_passsport: boolean;
        activites_passsport: string;
        accueil_handicap_moteur: boolean;
        accueil_handicap_mental: boolean;
    };
    coordonnees: {
        adresse_siege: {
            cplt_1?: string;
            cplt_2?: string;
            cplt_3?: string;
            num_voie?: number;
            type_voie?: string;
            voie?: string;
            bp?: string;
            cedex?: string;
            cp?: string;
            commune?: string;
            code_insee?: number;
        };
        adresse_siege_sirene?: {
            cplt_1?: string;
            cplt_2?: string;
            cplt_3?: string;
            num_voie?: number;
            type_voie?: string;
            voie?: string;
            bp?: string;
            cedex?: string;
            cp?: number;
            commune?: string;
            code_insee?: number;
        };
        adresse_gestion?: {
            cplt_1?: string;
            cplt_2?: string;
            cplt_3?: string;
            num_voie?: number;
            type_voie?: string;
            voie?: string;
            bp?: string;
            cedex?: string;
            cp?: number;
            commune?: string;
            code_insee?: number;
            pays?: string;
        };
        telephone?: string;
        courriel?: string;
        site_web?: string;
        publication_internet?: boolean;
    };
    reseau_affiliation: {
        id_rna: string;
        id_siren: number; // not tested since 2023-11-15 change
        nom: string;
        objet: string;
        adresse: string;
        telephone: string;
        courriel: string;
        nb_licencies: number;
        nb_licencies_h: number;
        nb_licencies_f: number;
        numero: string;
        url: string | number;
    }[];
    composition_reseau: {
        nom: string;
        id_rna: string;
        objet: string;
        id_siren: number;
        adresse: string;
    }[];
    adherent_personne_morales: {
        nom: string;
        siret: number;
        id: number;
    }[];
    representant_legal: StructureRepresentantLegalDto[];
    agrement: {
        type: string;
        numero: string;
        niveau: string;
        attributeur: string;
        date_attribution: string;
        url: string | number;
    }[];
    rh: {
        annee: number;
        nb_adherents: number;
        nb_adherents_h: number;
        nb_adherents_f: number;
        nb_benevoles: number;
        nb_volontaires: number;
        nb_salaries: number;
        nb_salaries_etpt: number;
        nb_emplois_aides: number;
        nb_personnels_detaches: number;
        cumul_top_5_salaires: number;
        id_siret: number;
    }[];
    compte: {
        annee: number;
        a_commissaire_aux_comptes: boolean;
        dons: number;
        subv: number;
        subv_cause: string;
        aides_3ans: number;
        charges: number;
        produits: number;
        resultat: number;
        id_siret: number;
    }[];
    etablissement: StructureEtablissementDto[];
    rib: StructureRibDto[];
    commentaire?: {
        note: string;
    };

    document_rna: StructureRnaDocumentDto[];
    document_dac: StructureDacDocumentDto[];
}
