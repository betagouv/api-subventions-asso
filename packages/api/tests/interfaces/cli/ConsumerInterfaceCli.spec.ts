import { ObjectId } from "mongodb";
import jwt from "jsonwebtoken";
import { JWT_SECRET } from "../../../src/configurations/jwt.conf";
import ConsumerCli from "../../../src/interfaces/cli/Consumer.cli";
import consumerTokenPort from "../../../src/dataProviders/db/user/consumer-token.port";
import userPort from "../../../src/dataProviders/db/user/user.port";
import { UserDto } from "dto";

describe("ConsumerCli", () => {
    const cli = new ConsumerCli();
    const EMAIL = "test@beta.gouv.fr";
    describe("create", () => {
        it("should create a user", async () => {
            await cli.create(EMAIL);
            const actual = await userPort.findByEmail(EMAIL);
            expect(actual).toMatchSnapshot({
                _id: expect.any(ObjectId),
                signupAt: expect.any(Date),
            });
        });

        it("should create a consumer token with user info", async () => {
            await cli.create(EMAIL);
            const user = (await userPort.findByEmail(EMAIL)) as UserDto;
            const token = (await consumerTokenPort.findToken(user._id)) as string;
            const actual = jwt.verify(token, JWT_SECRET);
            expect(actual).toMatchSnapshot({
                iat: expect.any(Number),
                _id: expect.any(String),
                signupAt: expect.any(String),
                now: expect.any(String),
            });
        });
    });
});
