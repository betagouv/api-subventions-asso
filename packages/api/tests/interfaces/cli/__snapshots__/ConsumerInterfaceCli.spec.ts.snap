// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConsumerCli create should create a consumer token with user info 1`] = `
Object {
  "_id": Any<String>,
  "active": false,
  "email": "test@beta.gouv.fr",
  "firstName": null,
  "iat": Any<Number>,
  "isConsumerToken": true,
  "lastActivityDate": null,
  "lastName": null,
  "now": Any<String>,
  "profileToComplete": true,
  "roles": Array [
    "user",
    "consumer",
  ],
  "nbVisits": 0,
  "signupAt": Any<String>,
}
`;

exports[`ConsumerCli create should create a user 1`] = `
Object {
  "_id": Any<ObjectId>,
  "active": false,
  "email": "test@beta.gouv.fr",
  "firstName": null,
  "lastActivityDate": null,
  "lastName": null,
  "profileToComplete": true,
  "roles": Array [
    "user",
    "consumer",
  ],
  "nbVisits": 0,
  "signupAt": Any<Date>,
}
`;
